pysr/regressor_sequence.py

```diff
@@ -0,0 +1,320 @@
+        if num_predictions < len(historical_X):
+            warnings.warn(
+                "The number of predictions is less than the number of historical data points. Some will be ignored."
+            )
+            historical_X = historical_X[:num_predictions]
+            return self._regressor.predict(X=historical_X, index=index)
+        else:
+            extra_predictions = num_predictions - len(historical_X)
+            pred = self._regressor.predict(X=historical_X, index=index)
+            for _ in range(extra_predictions):
+                pred_data = [pred[-self.recursive_history_length :].flatten()]
+                pred = np.concatenate(
+                    [pred, self._regressor.predict(X=pred_data, index=index)], axis=0
+                )
+            return pred
+
+    def _sliding_window(self, X):
+        return np.lib.stride_tricks.sliding_window_view(
+            X.flatten(), self.recursive_history_length * np.prod(X.shape[1])
+        )
+
+    @classmethod
+    def from_file(
+        cls,
+        *args,
+        recursive_history_length: int,
+        **kwargs,
+    ):
+        assert recursive_history_length is not None and recursive_history_length > 0
+
+        model = cls(recursive_history_length=recursive_history_length)
+        model._regressor = PySRRegressor.from_file(*args, **kwargs)
+        return model
+
+    def __repr__(self):
+        return self._regressor.__repr__().replace(
+            "PySRRegressor", "PySRSequenceRegressor", 1
+        )
+
+    def get_best(self, *args, **kwargs):
+        return self._regressor.get_best(*args, **kwargs)
+
+    def refresh(self, *args, **kwargs):
+        return self._regressor.refresh(*args, **kwargs)
+
+    def sympy(self, *args, **kwargs):
+        return self._regressor.sympy(*args, **kwargs)
+
+    def latex(self, *args, **kwargs):
+        return self._regressor.latex(*args, **kwargs)
+
+    def get_hof(self):
+        return self._regressor.get_hof()
+
+    def latex_table(
+        self,
+        *args,
+        **kwargs,
+    ):
+        """Create a LaTeX/booktabs table for all, or some, of the equations.
+
+        Parameters
+        ----------
+        indices : list[int] | list[list[int]]
+            If you wish to select a particular subset of equations from
+            `self.equations_`, give the row numbers here. By default,
+            all equations will be used. If there are multiple output
+            features, then pass a list of lists.
+        precision : int
+            The number of significant figures shown in the LaTeX
+            representations.
+            Default is `3`.
+        columns : list[str]
+            Which columns to include in the table.
+            Default is `["equation", "complexity", "loss", "score"]`.
+
+        Returns
+        -------
+        latex_table_str : str
+            A string that will render a table in LaTeX of the equations.
+        """
```

Comment:
```suggestion
```

Instead maybe the docstring can just link to the PySRRegressor one? Then there is no duplication.